### PR TITLE
Rolling Stock: initial checkin

### PR DESCRIPTION
**Pull request by roseundy**
First of many PRs.
Saving most of UI work until last

Common file change:
- Engine::Game::Meta - handle game names that don't have a number in them. Steam Over Holland will thank me.

### CODE REVIEW

File: lib/engine/game/g_rolling_stock/game.rb [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                        rose: '#b7274c',
+                        coral: '#f3716d',
+                        white: '#fff36b',
+                        navy: '#000080',
+                        cream: '#fffdd0',
+                        yellow: '#ffdea8')
+
+        CURRENCY_FORMAT_STR = '$%d'
+        BANK_CASH = 10_000
+        STARTING_CASH = 30
```
Comment (tobymao) [inline lib/engine/game/g_rolling_stock/game.rb:37]: 6 player game is 25

File: lib/engine/game/g_rolling_stock/game.rb [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        end
+
+        def new_investment_round
+          @log << "-- Turn #{@turn}, Phase 1 - Investment --"
+          @round_counter += 1
+          investment_round
+        end
+
+        def investment_round
+          GRollingStock::Round::Investment.new(self, [
```
Comment (tobymao) [inline lib/engine/game/g_rolling_stock/game.rb:137]: do you have to prefix with GRollingStock, isn't that implicit?

File: lib/engine/game/g_rolling_stock/entities.rb [hunk 3]
```diff
@@ -0,0 +1,467 @@
+# frozen_string_literal: true
+
+require_relative 'meta'
+require_relative '../base'
+
+module Engine
+  module Game
+    module GRollingStock
```
[the remark this answers is not in the document]
Comment (roseundy) [reply lib/engine/game/g_rolling_stock/entities.rb:8]: I figure both will be RollingStock and V1 will be an option.

File: lib/engine/game/g_rolling_stock/game.rb (hunk 1 above)
Comment (roseundy) [reply lib/engine/game/g_rolling_stock/game.rb:37]: Thanks. Missed that.

File: lib/engine/game/g_rolling_stock/game.rb (hunk 2 above)
Comment (roseundy) [reply lib/engine/game/g_rolling_stock/game.rb:137]: I thought I needed to, but apparently not.

**Pull request merged**